libs/partners/voyageai/langchain_voyageai/embeddings.py

```diff
@@ -105,22 +107,26 @@ def embed_documents(self, texts: List[str]) -> List[List[float]]:
 
         _iter = self._get_batch_iterator(texts)
         for i in _iter:
-            embeddings.extend(
-                self._client.embed(
-                    texts[i : i + self.batch_size],
-                    model=self.model,
-                    input_type="document",
-                    truncation=self.truncation,
-                ).embeddings
-            )
-
+            r = self._client.embed(
+                texts[i : i + self.batch_size],
+                model=self.model,
+                input_type="document",
+                truncation=self.truncation,
+                output_dimension=self.output_dimension,
+            ).embeddings
+            embeddings.extend(cast(Iterable[List[float]], r))
```

Comment:
why are all these casts necessary?

Reply:
As for the newest models (`voyage-3-large` and `voyage-code-3`) they also let you define the `output_dtype` and you can choose for example `float` or `int8`, `binary`. For anything that is not `float` we get an `int`:
1. https://github.com/voyage-ai/voyageai-python/blob/012d16bd708c797766beb26555e56b10db8d09fd/voyageai/object/embeddings.py#L8
2. https://github.com/voyage-ai/voyageai-python/blob/012d16bd708c797766beb26555e56b10db8d09fd/tests/test_client.py#L105

The default (as we use now) outputs `float`, but still I had to make this casting